fee-monitor/src/Stake.ts

```diff
@@ -0,0 +1,82 @@
+import { PlatformAddress, U64 } from "codechain-primitives/lib";
+import { sdk } from "./config";
+
+const RLP = require("rlp");
+
+function decodeU64(buffer: Buffer): U64 {
+    return U64.ensure("0x" + RLP.decode(buffer).toString("hex"));
+}
+
+function decodePlatformAddress(buffer: Buffer): PlatformAddress {
+    const accountId = buffer.toString("hex");
+    return PlatformAddress.fromAccountId(accountId, {
+        networkId: sdk.networkId,
+    });
+}
+
+export async function getStakeholders(blockNumber: number): Promise<PlatformAddress[]> {
+    const data = await sdk.rpc.engine.getCustomActionData(2, ["StakeholderAddresses"], blockNumber);
+    if (data == null) {
+        return [];
+    }
+    return RLP.decode(Buffer.from(data, "hex")).map(decodePlatformAddress);
+}
+
+async function getCCSBalance(address: PlatformAddress, blockNumber: number): Promise<U64> {
+    const data = await sdk.rpc.engine.getCustomActionData(
+        2,
+        ["Account", address.accountId.toEncodeObject()],
+        blockNumber,
+    );
+    if (data == null) {
```

Comment:
Because we request the CCS balance of stakeholders, it's an error if it's null.

Reply:
It can be a null according to the spec.